python/lsst/ci/hsc/gen2/validate.py

```diff
@@ -174,19 +174,7 @@ def checkPsfStarsAndFlags(self, catalog, minStellarFraction=0.95, doCheckFlags=T
         """Utility function for derived classes that want to verify PSF source
         selection and flag setting
         """
-        # pipe_task.propagateVisitFlags sets the flags PSF flag for
-        # all sources based on RA/DEC matching, so parent blends with
-        # children and failed blends are marked as PSF sources.
-        # There is also double counting when using scarlet, so we want
-        # to only check the undeblended isolated sources
-        # (parent sources with only one peak in their footprint)
-        # or the deblended models of isolated sources
-        # (the deblended child for each single peak parent source)
-        # but not both.
-        if "deblend_scarletFlux" in catalog.schema.getNames():
-            primary = catalog["parent"] != 0
-        else:
-            primary = catalog["deblend_nChild"] == 0
```

Comment:
So how is this different than what you get with `detect_isPrimary`?

Reply:
I can very much empathize with your confusion as I had to go through it in depth in the past (and can never remember all the nuances).  Long story short(-ish!) is that `scarlet` adds all kinds of complexity (compared with old deblender) and sets a slew of flags that are not trivial to parse (except for Fred!), making the determination of `detect_isPrimary` less-than-straight-forward.  Two main things have changed since Fred first added this accommodation for `scarlet`-based catalogs:
1. a lot of the flagging in `scarlet` changed
2. the `setPrimaryFlags` task in `pipe_tasks` was updated (on DM-28542 as part of the implementation of RFC-750) such that the `detect_isPrimary` flag always means what it's supposed to mean (regardless of which deblender was used), so if you want primary sources, select on that flag.  This is the update I'm making here (in principle this should've happened on DM-28542, but we missed it).

This second point is the important one and, if you want some gory details, see this [community post](https://community.lsst.org/t/new-use-of-isprimary-that-began-in-w-2021-12/4957).  I'm not going to even try to answer all your questions about the deleted comments (you'd actually need Fred for that, anyway) as the entire block has been moot since ~March 2021.